plugins/woocommerce/tests/legacy/unit-tests/crud/data.php

```diff
@@ -590,4 +590,192 @@ public function test_apply_changes_nested() {
+
+	/**
+	 * Test that __clone() properly clones meta_data objects in duplicate mode (default).
+	 */
+	public function test_clone_meta_data_objects_duplicate_mode() {
+		$object = $this->create_test_post();
+		$object->add_meta_data( 'test_meta_key', 'val1', true );
+		$object->add_meta_data( 'test_meta_key_2', 'val2', true );
+		$object->save_meta_data();
+
+		// Clone the object (default CLONE_MODE_DUPLICATE).
+		$cloned_object = clone $object;
+
+		// Get meta data from both objects.
+		$original_meta = $object->get_meta_data();
+		$cloned_meta   = $cloned_object->get_meta_data();
+
+		// Verify that the meta data arrays have the same count.
+		$this->assertCount( count( $original_meta ), $cloned_meta );
+
+		// Verify that the meta data objects are different instances (not shared references).
+		foreach ( $original_meta as $index => $meta ) {
+			$this->assertNotSame( $meta, $cloned_meta[ $index ], 'Meta data objects should not be the same instance after cloning' );
+			$this->assertNull( $cloned_meta[ $index ]->id, 'Meta IDs should be null in duplicate mode (backward compatible behavior)' );
+			$this->assertEquals( $meta->key, $cloned_meta[ $index ]->key, 'Meta keys should match' );
+			$this->assertEquals( $meta->value, $cloned_meta[ $index ]->value, 'Meta values should match' );
+		}
+
+		// Modify cloned meta and verify original is unchanged.
+		$cloned_meta[0]->value = 'modified_value';
+		$original_meta_after   = $object->get_meta_data();
+		$this->assertNotEquals( 'modified_value', $original_meta_after[0]->value, 'Modifying cloned meta should not affect original' );
+	}
+
+	/**
+	 * Test that __clone() preserves meta IDs in cache mode.
+	 */
+	public function test_clone_meta_data_objects_cache_mode() {
+		$object = $this->create_test_post();
+		$object->add_meta_data( 'test_meta_key', 'val1', true );
+		$object->add_meta_data( 'test_meta_key_2', 'val2', true );
+		$object->save_meta_data();
+
+		// Get original meta IDs.
+		$original_meta      = $object->get_meta_data();
+		$original_meta_id_1 = $original_meta[0]->id;
+		$original_meta_id_2 = $original_meta[1]->id;
+
+		// Set cache mode before cloning.
+		$object->set_clone_mode( WC_Data::CLONE_MODE_CACHE );
+
+		// Clone the object.
+		$cloned_object = clone $object;
+
+		// Get meta data from cloned object.
+		$cloned_meta = $cloned_object->get_meta_data();
+
+		// Verify that meta IDs are preserved in cache mode.
+		$this->assertEquals( $original_meta_id_1, $cloned_meta[0]->id, 'Meta IDs should be preserved in cache mode' );
+		$this->assertEquals( $original_meta_id_2, $cloned_meta[1]->id, 'Meta IDs should be preserved in cache mode' );
+
+		// Verify keys and values are preserved.
+		$this->assertEquals( 'test_meta_key', $cloned_meta[0]->key );
+		$this->assertEquals( 'val1', $cloned_meta[0]->value );
+		$this->assertEquals( 'test_meta_key_2', $cloned_meta[1]->key );
+		$this->assertEquals( 'val2', $cloned_meta[1]->value );
+
+		// Verify clone mode persists on cloned object.
```

Comment:
Can we _also_ add to this test the same "Modify cloned meta and verify original us unchanged." check here? The objects should _still_ be separate even though it's a straight clone!